src/lib/tree/Types.sol

```diff
@@ -1,20 +1,37 @@
 // SPDX-License-Identifier: Apache-2.0
 pragma solidity ^0.8.19;
 
-type NamespaceID is bytes29;
+/// @notice A representation of the Celestia-app namespace ID and its version.
+/// See: https://celestiaorg.github.io/celestia-app/specs/namespace.html
+struct Namespace {
+    // The namespace version.
+    bytes1 version;
+    // The namespace ID.
+    bytes28 id;
+}
+
+using {equalTo, lessThan, greaterThan, toBytes} for Namespace global;
 
-using {equality as ==} for NamespaceID global;
-using {lessthan as <} for NamespaceID global;
-using {greaterthan as >} for NamespaceID global;
+function equalTo(Namespace memory l, Namespace memory r) pure returns (bool) {
+    return l.toBytes() == r.toBytes();
+}
+
+function lessThan(Namespace memory l, Namespace memory r) pure returns (bool) {
+    return l.toBytes() < r.toBytes();
+}
 
-function equality(NamespaceID l, NamespaceID r) pure returns (bool) {
-    return NamespaceID.unwrap(l) == NamespaceID.unwrap(r);
+function greaterThan(Namespace memory l, Namespace memory r) pure returns (bool) {
+    return l.toBytes() > r.toBytes();
 }
 
-function lessthan(NamespaceID l, NamespaceID r) pure returns (bool) {
-    return NamespaceID.unwrap(l) < NamespaceID.unwrap(r);
+function toBytes(Namespace memory n) pure returns (bytes29) {
+    return bytes29(abi.encodePacked(n.version, n.id));
 }
 
-function greaterthan(NamespaceID l, NamespaceID r) pure returns (bool) {
-    return NamespaceID.unwrap(l) > NamespaceID.unwrap(r);
+function toNamespace(bytes29 n) pure returns (Namespace memory) {
```

Comment:
Hmmm. If you don't check for size here, would this allow an attacker to create a collision?

Reply:
the size is already 29, no? if I understand correctly, can't put a bigger bytes array in it

Reply:
Oh right, that's true.